src/Compilers/VisualBasic/Portable/GlobalImport.ImportDiagnosticInfo.vb

```diff
@@ -6,7 +6,7 @@ Imports System.Globalization
 
 Namespace Microsoft.CodeAnalysis.VisualBasic
 
-    Public Partial Class GlobalImport
+    Partial Public Class GlobalImport
```

Comment:
I'm fairly surprised by this ordering: the C# version is `internal partial`, not `partial internal`. This seems wrong.

Reply:
And, given the number of `Friend Partial`s that had to be changed in this PR, the rest of the codebase agrees. I'd prefer not do take this one.

Reply:
@333fred suprised me as well, but seems like this is the default order used by the code style option: https://github.com/dotnet/roslyn/blob/163404b1385780bee5fc8a19a9ffc8ca44fdb773/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/CodeStyle/VisualBasicCodeStyleOptions.vb#L23-L31

@CyrusNajmabadi any idea why the default orders diverge between C# and VB? C# order does not even include `partial` as I guess it needs to be the last one for the language?

https://github.com/dotnet/roslyn/blob/163404b1385780bee5fc8a19a9ffc8ca44fdb773/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpCodeStyleOptions.cs#L239-L250

Reply:
@333fred I agree that `Partial` shouldn't come before `Public`, `Private`, `Protected`, or `Friend`.
Not sure why that was the default.

I think updating the defaults will be a breaking change to others, so probably I should just set the preferred order in .editorconfig and not use the default?

Reply:
After `friend`, I think.

Reply:
@333fred I updated the order in editorconfig, and will do a FixAll for violations very soon.

Reply:
> C# order does not even include partial as I guess it needs to be the last one for the language?

@mavasani @CyrusNajmabadi previously, you couldn't have any other modifiers with `partial` in methods. Since that's changing C# 9, I think it's important that we address the ordering before 16.8 ships. In particular, things like `public partial async Task` ordering.

Reply:
> I think updating the defaults will be a breaking change to others, so probably I should just set the preferred order in .editorconfig and not use the default?

Can you file a separate design issue for this? Thanks.

Reply:
> 
> 
> > I think updating the defaults will be a breaking change to others, so probably I should just set the preferred order in .editorconfig and not use the default?
> 
> Can you file a separate design issue for this? Thanks.

Filed https://github.com/dotnet/roslyn/issues/47340

Reply:
> C# order does not even include partial as I guess it needs to be the last one for the language?

That was the reason initially.  HOwever, if we allow for partial-reordering at any point, we'd need to revisit this.

> @CyrusNajmabadi any idea why the default orders diverge between C# and VB? 

Yes.  Because they were separate teams and separate LSs.  C# (as you mentioned) required this to be last.  And VB shipped with their auto-cleanup set to put this first.  

So our editorconfig settings respect that as those are the patterns for the majority of our ecosystem.  not having Partial be first for VB is definitely an outlier here, and i would want to make our VB code consistent with teh VB experience around partial.